lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1398,6 +1404,99 @@ public function getSqlResultSetMappings()
         return $this->sqlResultSetMappings;
     }
 
+    /**
+     * Checks whether given property has type
+     *
+     * @param string $name Property name
+     *
+     * @return bool
+     */
+    private function isTypedProperty($name): bool
+    {
+        return PHP_VERSION_ID >= 70400
+               && isset($this->reflClass)
+               && $this->reflClass->hasProperty($name)
+               && $this->reflClass->getProperty($name)->hasType();
+    }
+
+    /**
+     * Validates & completes the given field mapping based on typed property.
+     *
+     * @param  array  $mapping The field mapping to validate & complete.
+     *
+     * @return void
+     */
+    private function _validateAndCompleteTypedFieldMapping(array &$mapping)
+    {
+        $type     = $this->reflClass->getProperty($mapping['fieldName'])->getType();
+
+        if ($type) {
+            if (! isset($mapping['nullable'])) {
+                $mapping['nullable'] = $type->allowsNull();
+            }
+
+            if (
+                ! isset($mapping['type'])
+                && ($type instanceof ReflectionNamedType)
+            ) {
+                switch ($type->getName()) {
+                    case DateInterval::class:
+                        $mapping['type'] = 'dateinterval';
+                        break;
+                    case DateTime::class:
+                        $mapping['type'] = 'datetime';
+                        break;
+                    case DateTimeImmutable::class:
+                        $mapping['type'] = 'datetime_immutable';
+                        break;
+                    case 'array':
+                        $mapping['type'] = 'json';
+                        break;
+                    case 'bool':
+                        $mapping['type'] = 'boolean';
+                        break;
+                    case 'float':
+                        $mapping['type'] = 'float';
+                        break;
+                    case 'int':
+                        $mapping['type'] = 'integer';
+                        break;
+                    case 'string':
+                        $mapping['type'] = 'string';
+                        break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Validates & completes the basic mapping information based on typed property.
+     *
+     * @param array $mapping The mapping.
+     *
+     * @return void
+     */
+    private function _validateAndCompleteTypedAssociationMapping(array &$mapping)
```

Comment:
same here, return the array again and don't pass by reference

Reply:
OK, done